riklet/src/core.rs

```diff
@@ -166,9 +172,24 @@ impl Riklet {
             if !Path::new(&file_path).exists() {
                 let lz4_path = format!("{}.lz4", &file_path);
                 fs::create_dir(&download_directory)?;
-                Self::download_image(&rootfs_url, &lz4_path)?;
 
-                Self::decompress(Path::new(&lz4_path), Path::new(&file_path))?;
+                match Self::download_image(&rootfs_url, &lz4_path) {
+                    Ok(_) => (),
+                    Err(e) => {
+                        log::error!("Error while downloading image: {}", e);
+                        fs::remove_dir_all(&download_directory)?;
+                        return Err(e);
+                    }
+                }
+
+                match Self::decompress(Path::new(&lz4_path), Path::new(&file_path)) {
+                    Ok(_) => (),
+                    Err(e) => {
+                        log::error!("Error while decompressing image: {}", e);
+                        fs::remove_dir_all(&download_directory)?;
+                        return Err(e);
+                    }
+                }
```

Comment:
Consider using `.map_err` instead
```suggestion
                Self::decompress(Path::new(&lz4_path), Path::new(&file_path)).map_err(|e| {
                        log::error!("Error while decompressing image: {}", e);
                        fs::remove_dir_all(&download_directory)?;
                        return Err(e);
                    }
                )
```

Reply:
This not works

the trait bound `Result<_, Box<dyn StdError>>: StdError` is not satisfied

Reply:
I guess the reason for this is because `fs::remove_dir_all()` has a Result which need to be handled. Below code is a working example

```rust
                Self::decompress(Path::new(&lz4_path), file_pathbuf).map_err(|e| {
                    log::error!("Error while decompressing image: {}", e);
                    fs::remove_dir_all(&download_directory)
                        .expect("Error while removing directory");
                    e
                })?;
```

Reply:
Done !